fontdrasil/src/orchestration.rs

```diff
@@ -0,0 +1,102 @@
+//! Common parts of work orchestration.
+
+use std::{
+    collections::HashSet,
+    fmt::Debug,
+    hash::Hash,
+    sync::{Arc, RwLock},
+};
+
+pub const MISSING_DATA: &str = "Missing data, dependency management failed us?";
+
+/// A unit of work safe to run in parallel
+///
+/// Naively you'd think we'd just return FnOnce + Send but that didn't want to compile
+/// See <https://github.com/rust-lang/rust/issues/29625>.
+///
+/// Data produced by work is written into a Context (type parameter C).
+pub trait Work<C, E> {
+    fn exec(&self, context: &C) -> Result<(), E>;
+}
+
+#[derive(Clone)]
+pub struct Cache<T: Default> {
+    pub item: Arc<RwLock<T>>,
+}
+
+impl<T: Default> Cache<T> {
+    pub fn new() -> Cache<T> {
+        Cache {
+            item: Default::default(),
+        }
+    }
+}
+
+impl<T: Default> Default for Cache<T> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+/// Tracks what parts of a context we can read/write.
+///
+/// Not meant to prevent malicious access, merely to detect mistakes
+/// as the result of mistaken concurrent access can be confusing to track down.
+pub struct Acl<I>
```

Comment:
what does "Acl" mean? "access control" or something?

Reply:
+1, would prefer a slightly more descriptive name.

Reply:
Interesting, I thought that name was very standard for this purpose. Will rename and expand comment.